### DISCUSSION TITLE
Associate a document page number to a partition element ?

### DISCUSSION THREAD

**Discussion by AntoninLeroy** [Unstructured-IO/unstructured#2528] [Q&A]
Hello,

I've been building some RAGs applications for a few months now with the help of this great package !
But there is something i didn't find. How could I associate to a partitioned element (title / text etc...) the original page number where the element is located within the document ?

The idea behind is to be able to store precisely where is the excerpt is located in the original document when vectorising my chunks and latter be able to tell the user where the LLM looked to construct it's answer.

Anyone worked on this ?

**Comment by Nandan9911** [comment]
Hi,

If you haven't yet found the solution for this then here's it : 

overide the setting of metadata in the partition() function by setting include_metadata=False in the function's argument. Then, once the partitioning is done,

from unstructured.documents.elements import ElementMetadata

metadata = ElementMetadata(
                    filename=file_name,
                    page_number=int(page_number),
                    file_directory=file_path,
                    languages=["auto"],
                )
element.metadata = copy.deepcopy(metadata)

This should solve your issue.
